### PR TITLE
[AMORO-2619]Skip the deleted direcotories when cleaning orphan files

### LINKED ISSUE #2619: [Bug]: Skip the deleted direcotories when cleaning orphan files
### What happened?

The exception and reasons are explained in #2579 . The only difference is that the current issue occurs in cleaning orphan files.

### Affects Versions

master

### What engines are you seeing the problem on?

AMS

### How to reproduce

_No response_

### Relevant log output

```shell
2024-03-05 12:49:29,322 ERROR [async-orphan-files-cleaning-executor-23] [com.netease.arctic.server.table.executor.OrphanFilesCleaningExecutor] [] - hms_ctripdi.dw_ctripdi.c_lp_ahead_show(tableId=8736605) failed to clean orphan file
java.lang.IllegalArgumentException: the target location(hdfs://ns/user/hive/warehouse/dw_ctripdi.db/c_lp_ahead_show/data/d=2023-11-13/meta_id_num_bucket=10) is not directory
        at org.apache.iceberg.relocated.com.google.common.base.Preconditions.checkArgument(Preconditions.java:143) ~[iceberg-bundled-guava-1.4.3.jar:?]
        at com.netease.arctic.io.ArcticHadoopFileIO.isEmptyDirectory(ArcticHadoopFileIO.java:161) ~[amoro-core-0.7.0-SNAPSHOT.jar:?]
        at com.netease.arctic.server.optimizing.maintainer.IcebergTableMaintainer.deleteInvalidFilesInFs(IcebergTableMaintainer.java:480) ~[amoro-ams-server-0.7.0-SNAPSHOT.jar:?]
        at com.netease.arctic.server.optimizing.maintainer.IcebergTableMaintainer.deleteInvalidFilesInFs(IcebergTableMaintainer.java:475) ~[amoro-ams-server-0.7.0-SNAPSHOT.jar:?]
        at com.netease.arctic.server.optimizing.maintainer.IcebergTableMaintainer.clearInternalTableContentsFiles(IcebergTableMaintainer.java:344) ~[amoro-ams-server-0.7.0-SNAPSHOT.jar:?]
        at com.netease.arctic.server.optimizing.maintainer.IcebergTableMaintainer.cleanContentFiles(IcebergTableMaintainer.java:290) ~[amoro-ams-server-0.7.0-SNAPSHOT.jar:?]
        at com.netease.arctic.server.optimizing.maintainer.IcebergTableMaintainer.cleanOrphanFiles(IcebergTableMaintainer.java:123) ~[amoro-ams-server-0.7.0-SNAPSHOT.jar:?]
        at com.netease.arctic.server.table.executor.OrphanFilesCleaningExecutor.execute(OrphanFilesCleaningExecutor.java:61) ~[amoro-ams-server-0.7.0-SNAPSHOT.jar:?]
        at com.netease.arctic.server.table.executor.BaseTableExecutor.executeTask(BaseTableExecutor.java:84) ~[amoro-ams-server-0.7.0-SNAPSHOT.jar:?]
        at com.netease.arctic.server.table.executor.BaseTableExecutor.lambda$null$2(BaseTableExecutor.java:74) ~[amoro-ams-server-0.7.0-SNAPSHOT.jar:?]
        at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511) [?:1.8.0_291]
        at java.util.concurrent.FutureTask.run(FutureTask.java:266) [?:1.8.0_291]
        at java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.access$201(ScheduledThreadPoolExecutor.java:180) [?:1.8.0_291]
        at java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.run(ScheduledThreadPoolExecutor.java:293) [?:1.8.0_291]
        at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149) [?:1.8.0_291]
        at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624) [?:1.8.0_291]
        at java.lang.Thread.run(Thread.java:748) [?:1.8.0_291]
```

### Anything else

_No response_

### Are you willing to submit a PR?

- [X] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

### PR DESCRIPTION
**Pull request by rfyu**
## Why are the changes needed?

Close #2619.

## Brief change log

-

## How was this patch tested?

- [ ] Add some test cases that check the changes thoroughly including negative and positive cases if possible

- [ ] Add screenshots for manual tests if appropriate

- [x] Run test locally before making a pull request

## Documentation

- Does this pull request introduce a new feature? (no)
- If yes, how is the feature documented? (not documented)

### CODE REVIEW

File: ams/server/src/main/java/com/netease/arctic/server/optimizing/maintainer/IcebergTableMaintainer.java [hunk 1]
```diff
@@ -485,7 +485,8 @@ private static int deleteInvalidFilesInFs(
       if (p.isDirectory()) {
         int deleted = deleteInvalidFilesInFs(fio, p.location(), lastTime, excludes);
         deleteCount += deleted;
-        if (!p.location().endsWith(METADATA_FOLDER_NAME)
+        if (fio.exists(p.location())
```
Comment (klion26) [inline ams/server/src/main/java/com/netease/arctic/server/optimizing/maintainer/IcebergTableMaintainer.java:488]: Do we need extract the logic to a function? seems the logic here is a bit complicated now.

Comment (XBaith) [reply ams/server/src/main/java/com/netease/arctic/server/optimizing/maintainer/IcebergTableMaintainer.java:488]: Makes sense, I think we could catch the exception in `clearInternalTableContentsFiles` and print the WRAN log to skip the unexpected error 😶

Comment (majin1102) [reply ams/server/src/main/java/com/netease/arctic/server/optimizing/maintainer/IcebergTableMaintainer.java:488]: > Do we need extract the logic to a function? seems the logic here is a bit complicated now.

This logic would be refactored (moved to optimizer) by process api, we could do this later

**Review by majin1102 — APPROVED** [review]
LGTM

**Pull request merged**